### PR TITLE
Updates and fixes to user guides and demos

### PR DESCRIPTION
**Pull request by philippjfr**
Fixed some issues in the dashboarding user guide and used tornado PeriodicCallback for asynchronous streaming data example.

### CODE REVIEW

**Comment by jlstevens** [comment]
I know you might still be working on this but I noticed that this is now in the guide ``renderer._figure_data``. If people are going to use this, do we need a suitable public interface?

**Comment by philippjfr** [comment]
>If people are going to use this, do we need a suitable public interface?

Yes probably.

File: examples/user_guide/15-Streaming_Data.ipynb [hunk 1]
```diff
@@ -568,29 +568,51 @@
...
     "buffer = Buffer(np.zeros((0, 2)), length=50)\n",
     "\n",
     "@gen.coroutine\n",
     "def f():\n",
-    "    for x in range(500):\n",
-    "        gen.sleep(0.1)\n",
-    "        buffer.send(np.array([[x, np.random.rand()]]))\n",
-    "        \n",
-    "IOLoop.current().add_callback(f)\n",
+    "    global count\n",
```
Comment (jlstevens) [inline examples/user_guide/15-Streaming_Data.ipynb]: Globals are the best thing to use here?

**Comment by jlstevens** [comment]
Looks fine though I would rather avoid that ``global`` if possible and we need a better API than ``_figure_data``. If there is an issue to remind us to improve the latter, I am happy to see this PR merged though I would be even happier if we could get rid of what looks like internal API calls before the release.

File: examples/user_guide/15-Streaming_Data.ipynb (hunk 1 above)
Comment (philippjfr) [reply examples/user_guide/15-Streaming_Data.ipynb]: Coroutines don't allow generators so that's not an option, open to other suggestions though.

**Comment by philippjfr** [comment]
>If there is an issue to remind us to improve the latter, I am happy to see this PR merged though I would be even happier if we could get rid of what looks like internal API calls before the release.

Not sure I want to add new API in this release and since this is just fixing broken docs I think we should merge and open an issue to discuss new API. I think there's more discussion to be had about the ``Renderer`` API anyway, particularly as it relates to bokeh server and perhaps providing some analogue to a ``show`` function.

**Comment by jlstevens** [comment]
Sounds like a plan. In that case, please make sure that issue exists, mentioning ``_figure_data`` and opening up discussion regarding the renderer API changes you would like to see.

**Comment by philippjfr** [comment]
Done in #2191.

**Comment by jlstevens** [comment]
Thanks. Otherwise the PR looks good.

Merging.

**Pull request merged**